### PR TITLE
Add cloud integration support

### PR DESCRIPTION
**Pull request by Ricky-Thomas**
Adding support for the following endpoints:

AWS
* GET (list)
* GET (list available namespaces)
* POST (create account config)
* DELETE (delete account config)
* PUT (generate external ID)
* PUT (update account)

AWS LOGS
* GET (list AWS logs integrations) 
* POST (add aws log lambda ARN)
* DELETE (delete lambda ARN config) 
* POST (aws integration logs check lambda async) - check if lambda_arn exists within account  
* GET (list log services)
* POST (AWS Integration Logs check Services Async) 
* POST (save services - this is what needs to run after the previous one to save the services they want for the recently added ARN)

AZURE ENDPOINTS
* GET (list all azure integrations)
* POST (create an azure integration)
* DELETE (delete an azure integration)
* POST (update host filters)
* PUT (update account)

GCP ENDPOINTS
* GET list
* POST create
* DELETE delete
* PUT (set GCP integration automute.) This also updates host filters. And is truly the only use case for updating a config here. It will allow you to update other fields as well, but not project_id and client_email which are the required fields. If those fields are to be updated an entirely new config should be created. We need to update the docs to reflect this info.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -75,6 +75,23 @@ module SpecDog
     end
   end
 
+  shared_examples 'an api method with params and body' do |command, args, request, endpoint, params, body|
```
Comment (mattsp1290) [inline spec/spec_helper.rb]: :shock:

File: lib/dogapi/facade.rb [hunk 2]
```diff
@@ -59,6 +59,7 @@ class Client # rubocop:disable Metrics/ClassLength
     attr_accessor :v2
     # Support for API version 2.
 
+    # rubocop:disable Metrics/MethodLength, Metrics/LineLength
```
Comment (mattsp1290) [inline lib/dogapi/facade.rb]: Overall looks good to me other than disabling some rubocop options. Is there a way to make this method work with these turned on? It looks like you may need to break out your initialization into some smaller sections that handle different bits of the setup. Like a cloud ints intialize that does aws, gcp and azure for example.

File: spec/spec_helper.rb (hunk 1 above)
Comment (mattsp1290) [reply spec/spec_helper.rb]: But also 👍

File: lib/dogapi/facade.rb (hunk 2 above)
Comment (Ricky-Thomas) [reply lib/dogapi/facade.rb]: Thanks Matt! I worked with Nick on this in working out this solution. At the time, we couldn't find a real good reason to limit the method to only 32 lines and also were not sure on a clean way to make those “line too long” errors in there shorter without adding a bunch of lines to method.

File: spec/spec_helper.rb (hunk 1 above)
Comment (Ricky-Thomas) [reply spec/spec_helper.rb]: 😄 thanks!

File: lib/dogapi/v1/aws_integration.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      #
+      # dog = Dogapi::Client.new(api_key, app_key)
+      #
+      # puts dog.generate_external_id(config)
+      def generate_external_id(config)
+        request(Net::HTTP::Put, "/api/#{API_VERSION}/integration/aws/generate_new_external_id", nil, config, true)
+      end
+
+      # Update integrated AWS account
+      # :config => Hash: integration config. NOT WORKING CURRENTLY
```
[the remark this answers is not in the document]
Comment (Ricky-Thomas) [reply lib/dogapi/v1/aws_integration.rb]: My mistake. This was a note to myself during the dev process as I'd been having a difficult time getting that one working. However, I was able to get it working and then forgot to remove the note. I've tested and double checked to make sure this is working properly. Will remove the note.

File: lib/dogapi/v1/aws_integration.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      def aws_list
+        request(Net::HTTP::Get, "/api/#{API_VERSION}/integration/aws", nil, nil, false)
+      end
+
+      # Create an AWS integration
+      # :config => Hash: integration config.
+      # config = {
+      #   :account_id => '<AWS_ACCOUNT>',
+      #   :host_tags => ['api:example'],
+      #   :role_name => '<AWS_ROLE_NAME>'
```
[the remark this answers is not in the document]
Comment (Ricky-Thomas) [reply lib/dogapi/v1/aws_integration.rb]: This has been added now. Thanks for flagging!

**Comment by nmuesch** [comment]
Closing this as its been superseded by https://github.com/DataDog/dogapi-rb/pull/201

**Pull request closed without merging**